eclipse-theia/theia
# Added Preferences UI widget

Signed-off-by: Nicholas Stenbeck <nicholas.stenbeck@ericsson.com>

#### What it does
This package adds a UI view to Preferences. It is a work in progress and is meant as a reference to get community approval, so that we can continue work in this direction. All feedback is welcome.
Addresses #6985.

##### Features
- Updates UI Similar to VS Code
- Fuzzy search to filter preferences
- Jump to category from navbar on the side
- Revert to default option from context menu
- Copy JSON code for preference from context menu

#### How to test
1. Go to file > settings > Open Preferences (UI).
2. Change a preference.
3. Go to file > settings > Open Preferences (JSON).
4. Confirm that the change is there.
5. Try writing something in the Preferences UI search bar and make sure that the preferences are filtered properly.
6. Click a category in the Preferences UI navbar and confirm that the Preferences UI widget scrolls to the right category.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
